File: app/src/main/java/com/anthonyessaye/opentv/Activities/DetailActivities/DetailActivity.java
package com.anthonyessaye.opentv.Activities.DetailActivities;

import android.os.Bundle;
import android.widget.Toast;

import androidx.core.content.ContextCompat;

import com.anthonyessaye.opentv.Activities.DetailsActivity;
import com.anthonyessaye.opentv.Persistence.Movie.Movie;
import com.anthonyessaye.opentv.R;

public class DetailActivity extends BaseTVActivity {

    private GlideBackgroundManager glideBackgroundManager;
    private Movie movie;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        movie = (Movie) this.getIntent().getSerializableExtra(DetailsActivity.MOVIE);
        DetailFragment detailsFragment = DetailFragment.newInstance(movie);
        addFragment(detailsFragment);

        glideBackgroundManager = new GlideBackgroundManager(this);
        updateBackground();
    }

    @Override
    protected void onResume() {
        super.onResume();
        updateBackground();
    }

    private void updateBackground() {
        if (movie != null) { //&& movie.getBackdropPath() != null) {
            Toast.makeText(this, "SOMETHING NEEDS TO BE DONE HERE", Toast.LENGTH_LONG).show();
            //glideBackgroundManager.loadImage(HttpClientModule.BACKDROP_URL + movie.getBackdropPath());
        } else {
            glideBackgroundManager.setBackground(ContextCompat.getDrawable(this, R.drawable.material_bg));
        }
    }
}

File: app/src/main/java/com/anthonyessaye/opentv/Activities/DetailActivities/DetailActivity.kt
package com.anthonyessaye.opentv.Activities.DetailActivities

import android.os.Bundle
import androidx.cardview.widget.CardView
import androidx.core.content.ContextCompat
import androidx.fragment.app.FragmentActivity
import com.anthonyessaye.opentv.Persistence.Movie.Movie
import com.anthonyessaye.opentv.R

class DetailActivity : FragmentActivity() {
    lateinit var cardView: CardView

    private var glideBackgroundManager: GlideBackgroundManager? = null
    private var movie: Movie? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        movie = this.getIntent().getSerializableExtra(DetailActivity.MOVIE) as Movie?

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.details_fragment, DetailFragment())
                .commitNow()
        }

        glideBackgroundManager = GlideBackgroundManager(this)
        updateBackground()
    }

    override fun onResume() {
        super.onResume()
        updateBackground()
    }

    private fun updateBackground() {
        glideBackgroundManager!!.setBackground(
            ContextCompat.getDrawable(
                this,
                R.drawable.material_bg
            )
        )
        if (movie != null) {// && movie.back != null) {
            //glideBackgroundManager.loadImage(HttpClientModule.BACKDROP_URL + movie.getBackdropPath());
        } else {
            glideBackgroundManager!!.setBackground(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.material_bg
                )
            )
        }
    }

    companion object {
        const val SHARED_ELEMENT_NAME = "hero"
        const val MOVIE = "Movie"
        const val LIVE_STREAM = "LIVE_STREAM"
    }
}
